pkg/didcomm/transport/ws/upgrade_srv.go

```diff
@@ -42,3 +43,20 @@ func acceptRecipient(pool *connPool, keys []string) bool {
 
 	return false
 }
+
+// keepConnAlive sends the pings the server based on time frequency. The web server, load balancer, network routers
+// between the client and server closes the TCP keepalives connection. This function calls websocket ping request
+// directly to the server and keeps the connection active.
+func keepConnAlive(conn *websocket.Conn, outbound bool, frequency time.Duration) {
+	if outbound {
+		for {
+			if err := conn.Ping(context.Background()); err != nil {
+				logger.Errorf("websocket ping error : %v", err)
+
+				break
+			}
+
+			time.Sleep(frequency)
```

Comment:
Shouldn’t we be using a timer/select pattern?

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/ab242869e92bb67c3cd0fd5bfbf474d1dd3db97a..873d9d4b0278aae40c2dad5a609270145a0332c1#diff-e3dec0159ab0c461e6c5bd749080659dR52) to for-range pattern.